src/Common/nunit/TestSelectionParserException.cs

```diff
@@ -21,46 +21,41 @@
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 // ***********************************************************************
 
-namespace NUnit.Framework.Internal.Filters
+using System;
+using System.Runtime.Serialization;
+
+#if NUNIT_ENGINE
+namespace NUnit.Engine
+#else
+namespace NUnit.Common
+#endif
 {
     /// <summary>
-    /// SimpleCategoryFilter parses a basic string representing a
-    /// single category or a list of categories separated by commas
+    /// TestSelectionParserException is thrown when an error 
+    /// is found while parsing the selection expression.
     /// </summary>
-    public class SimpleCategoryExpression
+#if !PORTABLE
+    [Serializable]
+#endif
+    public class TestSelectionParserException : Exception
     {
-        private string text;
-
-        private TestFilter filter;
-
         /// <summary>
-        /// Construct category filter from a text string
+        /// Construct with a message
         /// </summary>
-        /// <param name="text">A list of categories to parse</param>
-        public SimpleCategoryExpression(string text)
-        {
-            this.text = text;
-        }
+        public TestSelectionParserException(string message) : base(message) { }
 
         /// <summary>
-        /// Gets the TestFilter represented by the expression
+        /// Construct with a message and inner exception
         /// </summary>
-        public TestFilter Filter
-        {
-            get
-            {
-                if (filter == null)
```

Comment:
Lazy class does not exist in any version of .NET prior to 4.0.

Implementing it in the Compatibility classes might make sense, if it will significantly improve performance.

Not sure it will really make much difference?